### ISSUE TITLE
[Feature]: Computer Vision support - image robustness via pixel-level distortions

### ISSUE THREAD

**Issue by Adit-Jain-srm** [Adit-Jain-srm/NightmareNet#167]
## Problem

NightmareNet currently only supports NLP (text-based) distortions. The 4-phase Wake/Dream/Nightmare/Compress cycle is modality-agnostic in principle, but the distortion engines (`nightmarenet/distortions/`) only handle strings. No image/CV pipeline exists for adversarial image robustness training.

The paper (§5, §6.2 limitation #5) explicitly notes "Single dataset / model. SST-2 / DistilBERT only. Generalization to AG News, IMDB, BERT-base, GPT-2-class models is on the v2 roadmap" - extending to vision would be a major proof of the framework's generality.

**Relevant files:**
- `nightmarenet/distortions/registry.py` (plugin registry - vision distortions register here)
- `nightmarenet/distortions/base.py` (text distortion base - mirror this pattern for images)
- `nightmarenet/pipeline.py` (Pipeline orchestrator - needs to detect vision tasks)
- `nightmarenet/training/trainer.py` (Trainer - needs image DataLoader support)
- `nightmarenet/training/phases.py` (WakePhase, DreamPhase, NightmarePhase - need vision distortion application)

## Architecture

```
nightmarenet/distortions/vision/
├── __init__.py
├── base.py              # ImageDistortion ABC
├── dream.py             # Color jitter, geometric, blur, JPEG compression
├── nightmare.py         # FGSM, PGD, adversarial patches
└── registry.py          # Vision-specific registry (or extend existing)
```

Image representation: `torch.Tensor` (float32, shape `[C, H, W]`, values `[0, 1]`). Utility converters for `PIL.Image`.

## Sub-Issues

This parent issue is split into 4 sequential sub-issues. **Request assignment on THIS parent issue first**, then work through sub-issues in order:

1. **Base module** - `ImageDistortion` ABC + vision registry + Gaussian noise implementation
2. **Dream distortions** - Color jitter, geometric transforms, blur, JPEG compression
3. **Nightmare distortions** - FGSM, PGD attacks, adversarial patches (requires model access)
4. **Pipeline integration** - Wire vision distortions into the 4-phase training loop for image classification

Dependencies: 1 → 2 and 3 (parallel) → 4

## Assignment Process

To work on this feature:
1. Comment on THIS parent issue with your overall approach and timeline
2. Once assigned here, work through sub-issues sequentially (starting with 1/4)
3. Each sub-issue gets its own PR, merges independently

## Scope

Large (full feature). ~12-16 hours total across all 4 sub-issues. Requires familiarity with PyTorch vision models, torchvision transforms, and adversarial attack algorithms (FGSM/PGD).

---

> **Important:** Please read [CONTRIBUTING.md](https://github.com/Adit-Jain-srm/NightmareNet/blob/main/CONTRIBUTING.md) in full before requesting assignment. Assignment requests without a technical approach will not be considered. PRs demonstrating thorough research, benchmarks, or creative approaches beyond minimum criteria may earn bonus XP labels.

**Comment by Adit-Jain-srm** [comment]
**Assignment expectations for this feature:**

Requesting assignment here means committing to all 4 sub-issues (#168, #169, #170, #171) through to completion. Each gets its own independent PR that merges separately with ECSoC26 labels and eligible for bonus labels (`good-pr`, `good-backend`, `good-issue`) based on quality.

The dependency chain: #168 first → #169 and #170 can be parallel → #171 last.

What I'm looking for in your approach comment:
- Which image representation choices you'll make and why
- How you'll handle the gradient-based attacks in #170 (FGSM/PGD need model access during distortion)
- Your timeline for all 4 PRs
- Confirmation you have access to a GPU (needed for #170 and #171)

Don't request assignment if you can only do 1-2 of the sub-issues. This is a cohesive feature that benefits from one contributor understanding the full architecture end-to-end.

**Comment by Ruchira807** [comment]
@Adit-Jain-srm Would like to work on this issue under ECSOC26

**Comment by RitikaSen27** [comment]
Hi @Adit-Jain-srm can i get this assigned to me?
